internal/logging/sentry.go

```diff
@@ -2,28 +2,25 @@ package logging
 
 import (
 	"fmt"
-	"io"
 
 	"github.com/RHEnVision/provisioning-backend/internal/config"
 	sentrywriter "github.com/archdx/zerolog-sentry"
 	"github.com/getsentry/sentry-go"
+	"github.com/rs/zerolog"
 )
 
 // sentryWriter creates a zerolog writer for sentry.
 // Uses github.com/archdx/zerolog-sentry which is very simple wrapper.
-func sentryWriter(dsn string) (io.Writer, func(), error) {
-	sWriter, err := sentrywriter.New(dsn)
+func sentryWriter(dsn string) (zerolog.LevelWriter, func(), error) {
+	replacer := NewSentryReplacer()
+
+	sWriter, err := sentrywriter.New(dsn, sentrywriter.WithBeforeSend(replacer.Replace))
 	if err != nil {
 		return nil, func() {}, fmt.Errorf("cannot initialize sentry: %w", err)
 	}
 	sentry.ConfigureScope(func(scope *sentry.Scope) {
 		scope.SetTag("stream", config.BinaryName())
 	})
-	fWriter := NewSentryReplacer(sWriter)
 
-	closeFunc := func() {
-		_ = fWriter.Close()
-		_ = sWriter.Close()
-	}
-	return fWriter, closeFunc, nil
+	return sWriter, func() { _ = sWriter.Close }, nil
```

Comment:
I suggest rather to turn off this pesky warning in your IDE than this, Go compiler will not insist on this.